[NEXT] Cannot start nuxt: Cannot find module 'node_modules/vue-i18n/dist/vue-i18n.mjs'

Hi,

I'm currently using vue-i18n as a plugin in a Nuxt 3 project but I would like to switch to @nuxtjs/i18n as soon as possible. I tried to installed the alpha version but as soon as I declare the module in nuxt.config.js, I've got an error when I do npm run dev.

```
Cannot find module '.../node_modules/vue-i18n/dist/vue-i18n.mjs'

  at createEsmNotFoundErr (node:internal/modules/cjs/loader:983:15)
  at finalizeEsmResolution (node:internal/modules/cjs/loader:976:15)
  at resolveExports (node:internal/modules/cjs/loader:493:14)
  at Function.Module._findPath (node:internal/modules/cjs/loader:533:31)
  at Function.Module._resolveFilename (node:internal/modules/cjs/loader:942:27)
  at Function.resolve (node:internal/modules/cjs/helpers:108:19)
  at Function._resolve [as resolve] (node_modules/jiti/dist/jiti.js:1:243257)
  at resolveModule (node_modules/@nuxt/kit/dist/index.mjs:261:29)
  at setupAlias (node_modules/@nuxtjs/i18n/dist/module.mjs:25:36)
  at setup (node_modules/@nuxtjs/i18n/dist/module.mjs:541:11)
  at async Object.normalizedModule (node_modules/@nuxt/kit/dist/index.mjs:642:5)
  at async installModule (node_modules/nuxt/node_modules/@nuxt/kit/dist/index.mjs:456:3)
  at async initNuxt (node_modules/nuxt/dist/index.mjs:1606:7)
  at async load (node_modules/nuxi/dist/chunks/dev.mjs:6778:9)
  at async Object.invoke (node_modules/nuxi/dist/chunks/dev.mjs:6828:5)
  at async _main (node_modules/nuxi/dist/cli.mjs:50:20)
```

Indeed there is no such file in node_modules/vue-i18n/dist/
Any help will be appreciated!

I have the same issue on the 8.0.0-beta.1

The install is working but when i start the server he won't start on this error: 

> Cannot start nuxt:  Cannot find module '/Users/quentinguerrier/Desktop/seiz-io/oms/node_modules/vue-i18n/dist/vue-i18n.mjs'

This is in my package.json : 

`"@capacitor/browser": "^4.0.1",
    "@commitlint/cli": "^17.0.3",
    "@commitlint/config-conventional": "^17.0.3",
    "@commitlint/cz-commitlint": "^17.0.3",
    "@headlessui/vue": "^1.6.7",
    "@heroicons/vue": "^2.0.11",
    "@intlify/nuxt3": "^0.2.4",
    "@meforma/vue-toaster": "^1.3.0",
    "@nuxtjs/eslint-config-typescript": "^11.0.0",
    "@nuxtjs/google-fonts": "3.0.0-0",
    "@nuxtjs/strapi": "^1.5.1",
    "@nuxtjs/tailwindcss": "5.3.2",
    "commitizen": "^4.2.5",
    "cordova-res": "0.15.4",
    "daisyui": "^2.31.0",
    "eslint": "^8.23.1",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-prettier": "^4.2.1",
    "husky": "^8.0.1",
    "lodash": "^4.17.21",
    "moment": "^2.29.4",
    "nuxt": "3.0.0-rc.11",
    "prettier": "^2.7.1",
    "sass": "^1.54.5",
    "tailwindcss": "3.1.0",
    "v-calendar": "3.0.0-alpha.8",
    "vee-validate": "^4.6.9",
    "vue3-smooth-scroll": "^0.8.1",
    "yup": "^0.32.11"
    `

@kazupon No worries for the delay and thank you but you closed this thread based on the second error I mentioned. I tried again on a new project and yes the `Cannot read properties of undefined (reading 'path')` error is gone but not the `Cannot find module '.../node_modules/vue-i18n/dist/vue-i18n.mjs'` I declared for this thread when trying to install @nuxti18n in a project already working.

I can't share my project but maybe @ahoiroman , @LePtiDev or @smartfox87 have an example they can share please?

Otherwise I'll try to reproduce on a fresh project when I have some time, but I just saw that in a new project the vue-i18n.msj is there but not on my current project even if I'm doing the same command and config!

--- a/src/alias.ts
+++ b/src/alias.ts
@@ -1,6 +1,6 @@
 import createDebug from 'debug'
 import { resolvePath } from '@nuxt/kit'
-import { resolveVueI18nPkgPath, pkgModulesDir } from './dirs'
+import { pkgModulesDir } from './dirs'
 import { resolve } from 'pathe'
 import { VUE_I18N_PKG, VUE_I18N_BRIDGE_PKG, VUE_ROUTER_BRIDGE_PKG, VUE_I18N_ROUTING_PKG } from './constants'
 import { tryResolve, getPackageManagerType } from './utils'
@@ -15,7 +15,7 @@ export async function setupAlias(nuxt: Nuxt) {
   debug('setupAlias: pkgMgr', pkgMgr)
 
   // resolve vue-i18@v9
-  nuxt.options.alias[VUE_I18N_PKG] = await resolveVueI18nAlias(nuxt)
+  nuxt.options.alias[VUE_I18N_PKG] = await resolveVueI18nAlias(pkgModulesDir, nuxt, pkgMgr)
   nuxt.options.build.transpile.push(VUE_I18N_PKG)
   debug('vue-i18n alias', nuxt.options.alias[VUE_I18N_PKG])
 
@@ -25,55 +25,51 @@ export async function setupAlias(nuxt: Nuxt) {
   debug('@intlify/shared alias', nuxt.options.alias['@intlify/shared'])
 
   // resolve @intlify/vue-router-bridge
-  nuxt.options.alias[VUE_ROUTER_BRIDGE_PKG] = await resolveVueRouterBridgeAlias(
-    pkgModulesDir,
-    nuxt.options.rootDir,
-    nuxt.options.workspaceDir,
-    pkgMgr
-  )
+  nuxt.options.alias[VUE_ROUTER_BRIDGE_PKG] = await resolveVueRouterBridgeAlias(pkgModulesDir, nuxt, pkgMgr)
   nuxt.options.build.transpile.push(VUE_ROUTER_BRIDGE_PKG)
   debug('@intlify/vue-router-bridge alias', nuxt.options.alias[VUE_ROUTER_BRIDGE_PKG])
 
   // resolve @intlify/vue-i18n-bridge
-  nuxt.options.alias[VUE_I18N_BRIDGE_PKG] = await resolveVueI18nBridgeAlias(
-    pkgModulesDir,
-    nuxt.options.rootDir,
-    nuxt.options.workspaceDir,
-    pkgMgr
-  )
+  nuxt.options.alias[VUE_I18N_BRIDGE_PKG] = await resolveVueI18nBridgeAlias(pkgModulesDir, nuxt, pkgMgr)
   nuxt.options.build.transpile.push(VUE_I18N_BRIDGE_PKG)
   debug('@intlify/vue-i18n-bridge alias', nuxt.options.alias[VUE_I18N_BRIDGE_PKG])
 
   // resolve vue-i18n-routing
-  nuxt.options.alias[VUE_I18N_ROUTING_PKG] = await resolveVueI18nRoutingAlias(
-    pkgModulesDir,
-    nuxt.options.rootDir,
-    nuxt.options.workspaceDir,
-    pkgMgr
-  )
+  nuxt.options.alias[VUE_I18N_ROUTING_PKG] = await resolveVueI18nRoutingAlias(pkgModulesDir, nuxt, pkgMgr)
   nuxt.options.build.transpile.push(VUE_I18N_ROUTING_PKG)
   debug('vue-i18n-routing alias', nuxt.options.alias[VUE_I18N_ROUTING_PKG])
-}
-
-export async function resolveVueI18nAlias(nuxt: Nuxt) {
-  return resolve(await resolveVueI18nPkgPath(), nuxt.options.dev ? 'dist/vue-i18n.mjs' : 'dist/vue-i18n.runtime.mjs')
 }
 
 /**
  * NOTE:
  *  The following packages may not be able to resolve the file paths of the target ES modules with `resolvePath`
  *  so they resolve them on their own (sometimes, these are resolved as `cjs`)
+ *  - `vue-i18n`
  *  - `vue-i18n-routing`
  *  - `@intlify/vue-i18n-bridge`
  *  - `@intlify/vue-router-bridge`
  */
 
-async function resolveVueI18nBridgeAlias(
-  pkgModulesDir: string,
-  rootDir: string,
-  workspaceDir: string,
-  pkgMgr: PackageManager
-) {
+export async function resolveVueI18nAlias(pkgModulesDir: string, nuxt: Nuxt, pkgMgr: PackageManager) {
+  const { rootDir, workspaceDir } = nuxt.options
+  const modulePath = nuxt.options.dev
+    ? `${VUE_I18N_PKG}/dist/vue-i18n.mjs`
+    : `${VUE_I18N_PKG}/dist/vue-i18n.runtime.mjs`
+  const targets = [
+    // 1st, try to resolve from `node_modules` (hoisted case)
+    resolve(rootDir, 'node_modules', modulePath),
+    // 2nd, try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
+    resolve(pkgModulesDir, modulePath),
+    // workspace directories
+    resolve(workspaceDir, 'node_modules', modulePath)
+  ]
+  debug(`${VUE_I18N_PKG} resolving from ...`, targets)
+
+  return tryResolve(VUE_I18N_PKG, targets, pkgMgr)
+}
+
+async function resolveVueI18nBridgeAlias(pkgModulesDir: string, nuxt: Nuxt, pkgMgr: PackageManager) {
+  const { rootDir, workspaceDir } = nuxt.options
   const modulePath = `${VUE_I18N_BRIDGE_PKG}/lib/index.mjs` as const
   const targets = [
     // 1st, try to resolve from `node_modules` (hoisted case)
@@ -93,12 +89,8 @@ async function resolveVueI18nBridgeAlias(
   return tryResolve(VUE_I18N_BRIDGE_PKG, targets, pkgMgr)
 }
 
-async function resolveVueRouterBridgeAlias(
-  pkgModulesDir: string,
-  rootDir: string,
-  workspaceDir: string,
-  pkgMgr: PackageManager
-) {
+async function resolveVueRouterBridgeAlias(pkgModulesDir: string, nuxt: Nuxt, pkgMgr: PackageManager) {
+  const { rootDir, workspaceDir } = nuxt.options
   const modulePath = `${VUE_ROUTER_BRIDGE_PKG}/lib/index.mjs` as const
   const targets = [
     // 1st, try to resolve from `node_modules` (hoisted case)
@@ -118,12 +110,8 @@ async function resolveVueRouterBridgeAlias(
   return tryResolve(VUE_ROUTER_BRIDGE_PKG, targets, pkgMgr)
 }
 
-export async function resolveVueI18nRoutingAlias(
-  pkgModulesDir: string,
-  rootDir: string,
-  workspaceDir: string,
-  pkgMgr: PackageManager
-) {
+export async function resolveVueI18nRoutingAlias(pkgModulesDir: string, nuxt: Nuxt, pkgMgr: PackageManager) {
+  const { rootDir, workspaceDir } = nuxt.options
   const modulePath = `${VUE_I18N_ROUTING_PKG}/dist/vue-i18n-routing.mjs` as const
   const targets = [
     // 1st, try to resolve from `node_modules` (hoisted case)

--- a/src/dirs.ts
+++ b/src/dirs.ts
@@ -1,7 +1,6 @@
 import createDebug from 'debug'
 import { fileURLToPath } from 'node:url'
 import { dirname, resolve, parse as parsePath } from 'pathe'
-import { resolvePath } from '@nuxt/kit'
 import { VUE_I18N_ROUTING_PKG } from './constants'
 import { getPackageManagerType, tryResolve } from './utils'
 
@@ -16,12 +15,6 @@ debug('distDir', distDir)
 debug('runtimeDir', runtimeDir)
 debug('pkgDir', pkgDir)
 debug('pkgModulesDir', pkgModulesDir)
-
-export async function resolveVueI18nPkgPath() {
-  const p = await resolvePath('vue-i18n')
-  debug('vue-i18n resolved path', p)
-  return resolve(p, '../..')
-}
 
 export async function resolveVueI18nRoutingDtsPath(id: string, rootDir: string) {
   const pkgMgr = await getPackageManagerType()

--- a/src/module.ts
+++ b/src/module.ts
@@ -8,8 +8,8 @@ import { extendMessages } from './messages'
 import { extendBundler } from './bundler'
 import { generateLoaderOptions } from './gen'
 import { NUXT_I18N_MODULE_ID, DEFAULT_OPTIONS } from './constants'
-import { formatMessage, getNormalizedLocales, resolveLocales } from './utils'
-import { distDir, runtimeDir } from './dirs'
+import { formatMessage, getNormalizedLocales, resolveLocales, getPackageManagerType } from './utils'
+import { distDir, runtimeDir, pkgModulesDir } from './dirs'
 
 import type { NuxtI18nOptions } from './types'
 import type { DefineLocaleMessage, LocaleMessages } from 'vue-i18n'
@@ -204,7 +204,10 @@ export default defineNuxtModule<NuxtI18nOptions>({
      * auto imports
      */
 
-    const vueI18nPath = await resolveVueI18nAlias(nuxt)
+    const pkgMgr = await getPackageManagerType()
+    const vueI18nPath = await resolveVueI18nAlias(pkgModulesDir, nuxt, pkgMgr)
+    debug('vueI18nPath for auto-import', vueI18nPath)
+
     await addImports([
       { name: 'useI18n', from: vueI18nPath },
       ...[

